ewf.go

```diff
@@ -190,9 +212,23 @@ func NewWorkflow(name string, opts ...WorkflowOption) Workflow {
 		Steps:     []Step{},
 		State:     make(State),
 		CreatedAt: time.Now(),
+		Metadata:  make(map[string]string),
 	}
 	for _, opt := range opts {
 		opt(&w)
 	}
 	return w
 }
+
+func copyWorkflowMetadata(w *Workflow) {
```

Comment:
I think we just need to copy it one time after user input. no need to copy it after that